### PR TITLE
Fix multiz ventcrawling

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Allow mobs to travel through multiz adaptors,
fixes runtimes caused by ventcrawling

## Why It's Good For The Game
bug fix

## Changelog
:cl:
fix: allows ventcrawling between zlevels by using the multiz adapters and using the actions move_upwards and move_downwards
/:cl:

### CODE REVIEW

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 1]
```diff
@@ -475,29 +475,31 @@
...
-			var/list/pipenetdiff = returnPipenets() ^ target_move.returnPipenets()
-			if(pipenetdiff.len)
-				user.update_pipe_vision()
-			if(world.time - user.last_played_vent > VENT_SOUND_DELAY)
-				user.last_played_vent = world.time
-				playsound(src, 'sound/machines/ventcrawl.ogg', 50, TRUE, -3)
+
+	if(!target_move)
+		return
+	message_admins("PASS TARGET")
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:487]: I'm assuming you forgot about these?

File: code/modules/atmospherics/machinery/atmosmachinery.dm [hunk 2]
```diff
@@ -475,29 +475,31 @@
 
 // Handles mob movement inside a pipenet
 /obj/machinery/atmospherics/relaymove(mob/living/user, direction)
-	direction &= initialize_directions
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/machinery/atmosmachinery.dm:478]: why was this removed?

File: code/modules/mob/mob_movement.dm [hunk 3]
```diff
@@ -496,24 +496,27 @@
...
+	if(zMove(UP, TRUE, ventcrawling_mob))
 		to_chat(src, "<span class='notice'>You move upwards.</span>")
 
 ///Moves a mob down a z level
 /mob/verb/down()
 	set name = "Move Down"
 	set category = "IC"
 
-	if(zMove(DOWN, TRUE))
+	var/ventcrawling_mob = HAS_TRAIT(src, TRAIT_MOVE_VENTCRAWLING)
```
Comment (LemonInTheDark) [inline code/modules/mob/mob_movement.dm:513]: what?

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 1 above)
Comment (Ghilker) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:487]: derp lol

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 2 above)
Comment (Ghilker) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:478]: no idea why was added in the first place, this should check if the direction we are moving to has a connected machine. I did all the tests and this line is totally not needed

File: code/modules/mob/mob_movement.dm (hunk 3 above)
Comment (Ghilker) [reply code/modules/mob/mob_movement.dm:513]: to be able to move up and down while ventcrawling we need to check if the mob has that trait, so that when the proc checks for zMove down, it will pass that bool to the proc

Comment (LemonInTheDark) [reply code/modules/mob/mob_movement.dm:513]: why is it called ventcrawling mob? it's not the ventcrawling mob

Comment (Ghilker) [reply code/modules/mob/mob_movement.dm:513]: Is more like a question "is the mob doing this movement vent crawling?"

File: code/modules/atmospherics/machinery/atmosmachinery.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:478]: Doesn't this line prevent moving into something that you don't share a corner with? Does it ever cause a return if you keep it there?

Comment (Ghilker) [reply code/modules/atmospherics/machinery/atmosmachinery.dm:478]: i just tested it and readding the line breaks the code and causes multiple not needed calls (since the smart pipes are ALL_CARDINAL you check all sides when moving) and also you can't move up and down when it's present
